### PR TITLE
add imaging metadata

### PR DESCRIPTION
**Pull request by alessandratrapani**

### CODE REVIEW

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023segmentationinterface.py [hunk 1]
```diff
@@ -20,6 +22,59 @@ def __init__(
...
+            verbose=verbose,
+        )
+        self.segmentation_extractor = self.segmentation_extractor.frame_slice(
+            start_frame=start_frame, end_frame=end_frame
+        )
+
+    def get_metadata(self) -> DeepDict:
+        metadata = super().get_metadata()
+
+        device_name = "CustomMicroscope"  # TODO how can I use the yaml file spec here?
```
Comment (weiglszonja) [inline src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023segmentationinterface.py]: If you're updating the metadata with the editable metadata (yaml) in the `convert_session()` script, then I think here at `metadata = super().get_metadata()` you should see two devices, the default device and the "CustomMicroscope"?

Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023segmentationinterface.py:39]: Then I am doing something wrong because I see only the default device

Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023segmentationinterface.py]: I only see Microscope for some reason...

**Comment by alessandratrapani** [comment]
It worked, thanks @weiglszonja ! example here: https://catalystneuro-processing.s3.us-east-2.amazonaws.com/adesnik-lab-to-nwb/5stim.nwb

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023segmentationinterface.py [hunk 2]
```diff
@@ -20,6 +22,59 @@ def __init__(
...
+            channel_name=channel_name,
+            plane_name=plane_name,
+            plane_segmentation_name=plane_segmentation_name,
+            verbose=verbose,
+        )
+        self.segmentation_extractor = self.segmentation_extractor.frame_slice(
+            start_frame=start_frame, end_frame=end_frame
+        )
+
+    # def get_metadata(self) -> DeepDict:
```
Comment (weiglszonja) [inline src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023segmentationinterface.py]: Can you remove if not necessary?

Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023segmentationinterface.py]: Ah yeah! thanks

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py [hunk 3]
```diff
@@ -68,44 +68,49 @@ def get_metadata(self) -> DeepDict:
         # Extract many scan image properties and attach them as dict in the description
         if self.image_metadata is not None:
             extracted_description = json.dumps(self.image_metadata)
-            two_photon_series_metadata.update(description=extracted_description)
+            two_photon_series_metadata.update(comments=extracted_description)
```
Comment (weiglszonja) [inline src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: I just looked at the file, and the comments look really weird. I see the intention to keep as much information as possible but this is too much:

Screenshot 2024-01-26 at 12 47 11

I think a better way to store all this information is either to create an extension for this, if that is out of scope than add it as a DynamicTable. 

Let me know what you think

**Review by weiglszonja — APPROVED** [review]
Besides the comment that I had on "comment" :D the file looks good, thanks!

Don't forget to delete the test files as they are 3GB each.

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py (hunk 3 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: I agree, but I also think it's not worth to keep all of this info. I think I will just remove from comments.
